File: app/src/main/java/chenmc/sms/ui/main/PreferenceActivity.java
package chenmc.sms.ui.main;

import android.os.Build;
import android.os.Bundle;

import chenmc.sms.code.helper.R;
import chenmc.sms.ui.app.PermissionActivity;
import chenmc.sms.ui.interfaces.IOnBackPressedActivity;
import chenmc.sms.ui.interfaces.IOnBackPressedFragment;

/**
 * @author 明 明
 *         Created on 2017-2-13.
 */

public class PreferenceActivity extends PermissionActivity implements IOnBackPressedActivity {
    
    private IOnBackPressedFragment mOnBackPressedFragment;
    
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        if (getActionBar() != null && Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
            getActionBar().setElevation(0);
        }
        
        getFragmentManager().beginTransaction()
            .replace(R.id.fragment_container, new MainPreferenceFragment())
            .commit();
    }
    
    @Override
    protected int getContentViewRes() {
        return R.layout.activity_preference;
    }
    
    @Override
    public void setFocusFragment(IOnBackPressedFragment fragment) {
        mOnBackPressedFragment = fragment;
    }
    
    @Override
    public void onBackPressed() {
        if (mOnBackPressedFragment == null || !mOnBackPressedFragment.onBackPressed()) {
            super.onBackPressed();
        }
    }
}

File: app/src/main/java/chenmc/sms/ui/main/PreferenceActivity.kt
package chenmc.sms.ui.main

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import chenmc.sms.code.helper.R
import chenmc.sms.ui.interfaces.OnBackPressedActivity
import chenmc.sms.ui.interfaces.OnBackPressedFragment

/**
 * @author 明 明
 * Created on 2017-2-13.
 */

class PreferenceActivity : AppCompatActivity(), OnBackPressedActivity {

    private var mOnBackPressedFragment: OnBackPressedFragment? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_preference)

        supportFragmentManager.beginTransaction()
            .replace(R.id.fragment_container, MainPreferenceFragment())
            .commit()
    }

    override fun setFocusFragment(fragment: OnBackPressedFragment) {
        mOnBackPressedFragment = fragment
    }

    override fun onBackPressed() {
        if (mOnBackPressedFragment?.onBackPressed() != true) {
            super.onBackPressed()
        }
    }
}
